ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/AbstractTaskFragment.kt

```diff
@@ -145,16 +149,27 @@ abstract class AbstractTaskFragment<T : AbstractTaskViewModel> : AbstractFragmen
 
   protected fun addButton(action: ButtonAction): TaskButton {
     check(!buttons.contains(action)) { "Button $action already bound" }
+    val updatedAction = maybeOverrideButton(action)
     val button =
       TaskButtonFactory.createAndAttachButton(
-        action,
+        updatedAction,
         taskView.actionButtonsContainer,
         layoutInflater
       )
-    buttons[action] = button
+    buttons[updatedAction] = button
     return button
   }
 
+  /**
+   * Changes the button from "Next" to "Done" if the current task fragment is last in it's position.
+   */
+  private fun maybeOverrideButton(action: ButtonAction): ButtonAction =
```

Comment:
Can this be done reactively / declaratively using data binding rather than programmatically? e.g. by adding LiveData<Boolean> `isLastTask` to the ViewModel?

Reply:
I don't think we can do it reactively or using data binding. We create the layout of the task fragment dynamically

Reply:
I see. Sgtm!